Automattic/jetpack
# Search: Add E2E tests for search dashboard

#### Changes proposed in this Pull Request:
The PR added E2E tests for the search dashboard, including two happy flows,

1. The page displays all elements right
2. User can toggle module and instant search setting
    1. User can toggle off search module, and instant search is toggled off automatically
    2. User can toggle on instant search, and the search module would be turned on automatically 
    3. User can toggle on/off instant search
        1. Buttons are disabled Regardingly

#### Jetpack product discussion
PcNPJE-uw-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
Setup and build: 
* `cd projects/plugins/jetpack/tests/e2e`
* `pnpm install`
* `pnpx jetpack build packages/search`
* `pnpx jetpack build plugins/jetpack`
* `pnpm env-start`
* `pnpm tunnel-on`
Run the dashboard E2E tests:
* `pnpm test-e2e -- ./specs/post-connection/search-dashboard.test.js`
* Ensure all tests pass

Note: you can add `--headed` at the end of the command to run Chromium in the foreground.
